src/client/components/UserPage/Widgets/CopyButton.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+import copy from 'copy-to-clipboard'
+import { Button, Typography, makeStyles, createStyles } from '@material-ui/core'
+
+import copyIcon from '../assets/copy-icon.svg'
+import OnClickTooltip from './OnClickTooltip'
+
+type StyleProps = {
+  iconSize: number
+}
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    copyLinkDiv: {
+      display: 'flex',
+    },
+    copyIcon: {
+      marginRight: 5,
+      width: (props: StyleProps) => props.iconSize,
+    },
+  }),
+)
+
+export type CopyButtonProps = {
+  shortUrl: string
+  buttonText: string
+  iconSize: number
+  variant?:
+    | 'button'
+    | 'caption'
+    | 'h1'
+    | 'h2'
+    | 'h3'
+    | 'h4'
+    | 'h5'
+    | 'h6'
+    | 'inherit'
+    | 'subtitle1'
+    | 'subtitle2'
+    | 'body1'
+    | 'body2'
+    | 'overline'
+    | 'srOnly'
+    | undefined
```

Comment:
wow this variant thing is huge, also gives the impression that it'll eventually be reused at some place since other html elements might have this variant thing(?). consider if it's worth extracting this into a variant enum of some sort

Reply:
Sounds good, shall do that!